website/src/hooks/ui/useUserScore.ts

```diff
@@ -0,0 +1,18 @@
+import { useSession } from "next-auth/react";
+import { get } from "src/lib/api";
+import { LeaderboardEntity, LeaderboardTimeFrame } from "src/types/Leaderboard";
+import uswSWRImmutable from "swr/immutable";
+
+export const useUserScore = () => {
+  const { status } = useSession();
+  const isLoggedIn = status === "authenticated";
+  const { data: entries } = uswSWRImmutable<Partial<{ [time in LeaderboardTimeFrame]: LeaderboardEntity }>>(
+    isLoggedIn && "/api/user_stats",
+    get,
+    {
+      dedupingInterval: 1000 * 60, // once per minute
```

Comment:
This should be `refreshInterval` right? Also can you add `keepPreviousData` to avoid flashing UI?

Reply:
not really, I am using `useSWRImmutable` since the user scores are updated once every hour or so (if I am not mistaken), so there is no need to fetch new data.